test/screenshot.spec.ts

```diff
@@ -50,15 +50,15 @@ describe('Screenshots', function () {
       });
       expect(screenshot).toBeGolden('screenshot-clip-rect.png');
     });
-    itFailsFirefox('should clip elements to the viewport', async () => {
+    itFailsFirefox('should clip elements to the viewport size', async () => {
       const { page, server } = getTestState();
 
-      await page.setViewport({ width: 500, height: 500 });
+      await page.setViewport({ width: 50, height: 50 });
       await page.goto(server.PREFIX + '/grid.html');
       const screenshot = await page.screenshot({
         clip: {
-          x: 50,
-          y: 600,
+          x: 0,
```

Comment:
why would not the old test work?

Reply:
According to what I got from the [CL 2643792](crrev.com/c/2643792), it used to cut the screenshot respecting the ViewPort position, but now the viewport moves to the clip first, and then taking screenshot. So the initial ViewPort offset doesn't matter anymore.
Actual output of the initial test: https://i.imgur.com/QC2rTB5.png

Reply:
Corresponding bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1164610

Reply:
So this is a breaking change in how Puppeteer's `clip` works?

Reply:
yes, it is. 
Another breaking change is in the `AriaQuery`

Reply:
Mentioned in the initial commit.

Reply:
Oh, it's the breaking change in how the `screenshot` works. The `clip` is still respected in both offset and size. But the result screenshot is filled with white without respect of the initial ViewPort offset, only based on its size.

Reply:
BTW, there are some problems with the respecting `offset` in the `screenshot`. 
EG this example produces the incorrectly-looking result: https://i.imgur.com/syIImrg.png
```
await page.setViewport({ width: 50, height: 50 });
await page.goto(server.PREFIX + '/grid.html');
const screenshot = await page.screenshot({
  clip: {
    x: 25,
    y: 25,
    width: 100,
    height: 100,
  },
});
```

Reply:
Okay, thanks. Btw, per https://www.conventionalcommits.org/en/v1.0.0/#summary we should use this format:

```
BREAKING CHANGE: …
```